ckanext/querytool/fanstatic/javascript/vitals.js

```diff
@@ -1456,3 +1456,127 @@ $(document).ready(function(){
 });
+
+document.addEventListener("DOMContentLoaded", function() {
+  const send2faEmailBtn = document.getElementById("send-2fa-email");
+
+  if (send2faEmailBtn) {
+    send2faEmailBtn.addEventListener("click", function(event) {
+      const mfaField = document.getElementById("mfa-email");
+      mfaField.style.display = "block";
+      const existingSuccessMessage = document.querySelector(
+        ".alert-success"
+      );
+
+      if (existingSuccessMessage) {
+        existingSuccessMessage.remove();
+      }
+
+      const existingErrorMessage = document.querySelector(
+        ".alert-danger"
+      );
+
+      if (existingErrorMessage) {
+        existingErrorMessage.remove();
+      }
+
+      event.preventDefault();
+      const username = document.getElementById("field-login").value;
+
+      $.ajax({
+        url: `/send_2fa_code/${username}`,
+        type: "POST",
+        success: function(response) {
+          const successMessage = document.createElement("p");
+
+          successMessage.innerHTML = 'A verification code has been sent. This code expires in: '
+          successMessage.classList.add("alert", "alert-success");
+          successMessage.style.marginTop = "1rem";
+          $(successMessage).css("margin-left", "130px");
+          $(successMessage).css("width", "320px");
+
+          successMessage
+
+          const countdownTimer = createCountdownTimer(600);
+          successMessage.appendChild(countdownTimer);
+
+          const twofaEmailLink = document.getElementById("mfa-email-link");
+          twofaEmailLink.parentNode.insertBefore(
+            successMessage,
+            twofaEmailLink.nextSibling
+          );
+        },
+        error: function(error) {
+          const errorMessage = document.createElement("p");
+
+          errorMessage.innerHTML = 'There was an error sending the verification code. Please try again.'
+          errorMessage.classList.add("alert", "alert-danger");
+          errorMessage.style.marginTop = "1rem";
+          errorMessage.style.marginLeft = "130px;";
+
+          const twofaEmailLink = document.getElementById("mfa-email-link");
+
+          twofaEmailLink.parentNode.insertBefore(
+            errorMessage,
+            twofaEmailLink.nextSibling
+          );
+        }
+      });
+
+      const twofaEmailLink = document.getElementById("send-2fa-email");
+      twofaEmailLink.innerHTML = "Resend verification code";
+    });
+  }
+});
+
+document.addEventListener("DOMContentLoaded", function() {
+  const loginField = document.getElementById("field-login");
+  const passwordField = document.getElementById("field-password");
+  const send2faEmailBtn = document.getElementById("mfa-email-link");
+  const mfaField = document.getElementById("field-mfa");
+
+  if (loginField && passwordField && send2faEmailBtn) {
+    loginField.addEventListener("input", function(event) {
+      if (loginField.value && passwordField.value) {
+        send2faEmailBtn.style.display = "block";
```

Comment:
@mpolidori Actually I think this is not a good idea, because when the browser autocompletes the credentials the button continues invisible (or check field on load once)

![ss-2023-05-10-18:22:43](https://github.com/datopian/ckanext-querytool/assets/16550934/2cc91603-321d-4e52-9f37-50374e0622dd)